hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/metrics/UserMetricsImp.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.metrics;
+
+import com.hazelcast.jet.core.metrics.Counter;
+
+public final class UserMetricsImp {
+
+    private static final ThreadLocal<UserMetricsContext> CONTEXT = new ThreadLocal<>();
+
+    private UserMetricsImp() {
+    }
+
+    public static void set(UserMetricsContext context) {
+        CONTEXT.set(context);
+    }
+
+    public static Counter get(String name) {
+        UserMetricsContext context = CONTEXT.get();
+        return context.get(name);
```

Comment:
We should throw some more informative exc if `context` is null, like "This thread is not a processor thread, metrics can be used only on processor threads"